app/lib/actions/remote_execution/run_hosts_job.rb

```diff
@@ -32,6 +31,8 @@ def plan(job_invocation)
         set_up_concurrency_control job_invocation
         input.update(:job_category => job_invocation.job_category)
         plan_self(:job_invocation_id => job_invocation.id)
+        trigger_action = plan_action(Actions::TriggerProxyBatch, batch_size: proxy_batch_size, total_count: hosts.count)
+        input[:trigger_run_step_id] = trigger_action.run_step_id
```

Comment:
What happens if batch triggering is turned off? Will `TriggerProxyBatch` ever finish?

Reply:
Yeah, it will do nothing and the last `TriggerLastBatch` event will cause it to finish.